File: codeforces/42.amusing_joke_[141A]/amusing_joke.py
# Author: José Rodolfo (jric2002)
name_guest = str(input())
name_host = str(input())
letters = []
letters[:0] = str(input())
text = name_guest + name_host
for i in range(0, len(text)):
  exist = False
  for j in range(0, len(letters)):
    if (text[i] == letters[j]):
      letters.pop(j)
      exist = True
      break
  if (exist != True):
    break
if ((len(letters) == 0) and exist):
  print("YES")
else:
  print("NO")

File: codeforces/low_difficulty/800_points/42.amusing_joke_[141A]/amusing_joke.cpp
#include <iostream>
#include <cstring>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  string name_guest, name_host, text, letters;
  bool exist;
  cin >> name_guest;
  cin >> name_host;
  cin >> letters;
  text = name_guest + name_host;
  for (unsigned short int i = 0; i < text.size() && !letters.empty(); i++) {
    unsigned short int j = 0;
    exist = false;
    while (j < letters.size() && !letters.empty()) {
      if (text[i] == letters[j]) {
        exist = true;
        letters.erase(j, 1);
        break;
      }
      j++;
    }
    if (!exist) {
      break;
    }
  }
  if (letters.empty() && exist) {
    cout << "YES" << endl;
  }
  else {
    cout << "NO" << endl;
  }
  return 0;
}